dissect/target/plugins/os/windows/network.py

```diff
@@ -0,0 +1,383 @@
+import datetime
+
+from dissect.target.exceptions import (
+    RegistryKeyNotFoundError,
+    RegistryValueNotFoundError,
+)
+from dissect.target.helpers.record import WindowsInterfaceRecord
+from dissect.target.plugin import internal
+from dissect.target.plugins.general.network import NetworkPlugin
+
+WINDOWS_EPOCH = datetime.datetime(1601, 1, 1)
+UNIX_EPOCH = datetime.datetime(1970, 1, 1)
+EPOCH_DIFF = (UNIX_EPOCH - WINDOWS_EPOCH).total_seconds()  # Difference in seconds
+
+
+class WindowsNetworkPlugin(NetworkPlugin):
+    @internal
+    def _interfaces(self) -> list[WindowsInterfaceRecord]:
+        for device_info in self._extract_network_devices():
+            yield WindowsInterfaceRecord(
+                _target=self.target,
+                dns=device_info["dns"],
+                search_domain=device_info["domain"],
+                gateway=device_info["gateway"],
+                ip=device_info["ip"],
+                first_connected=device_info["NetworkInterfaceInstallTimestamp"],
+                mac=device_info.get("NetworkAddress", None),
+                metric=device_info["metric"],
+                name=device_info["Name"],
+                source="registry",
+                subnetmask=device_info["subnetmask"],
+                type=device_info["Type"],
+                vlan=device_info.get("Vlan", None),
+                last_connected=None,
+                enabled=None,
+            )
+
+    def _extract_network_devices(self) -> list[dict]:
+        # Windows registry for Network Adapter information
+        key = "HKLM\\SYSTEM\\CurrentControlSet\\Control\\Class\\{4d36e972-e325-11ce-bfc1-08002be10318}"
+        fields = [
+            "NetCfgInstanceId",
+            "NetworkAddress",
+            "NetworkInterfaceInstallTimestamp",
+            "DriverDesc",
+            "*PriorityVlanTag",
+            "*IfType",
+        ]
+
+        for r in self.target.registry.keys(key):
+            for s in r.subkeys():
+                device_info = {}
+                config = None
+
+                for field in fields:
```

Comment:
If you're going to access every field anyway and check for it with an `if`, I'd not use a loop and just hardcode every field access.